### ISSUE TITLE
`no-any-return` dependent on order of conditional

### ISSUE THREAD

**Issue by jamiesandenr** [python/mypy#9631]
**Bug Report**

Whether a `no-any-return` error is detected or not in a ternary operator or conditional statement depends on which way round the cases are written.  Furthermore, in the ternary operator case it's not caught at all if the result is immediately returned (rather than assigned to a variable).

**To Reproduce**

```python
anything: Any

##### Ternary operator with assignment


def foo_ternary_assign(cond: bool) -> Optional[int]:
    x = None if cond else anything
    return x

# error
def bar_ternary_assign(cond: bool) -> Optional[int]:
    x = anything if cond else None
    return x


##### Ternary operator without assignment


def foo_ternary_return(cond: bool) -> Optional[int]:
    return None if cond else anything


def bar_ternary_return(cond: bool) -> Optional[int]:
    return anything if cond else None


##### Conditional statement with assignment


def foo_stmt_assign(cond: bool) -> Optional[int]:
    if cond:
        x = None
    else:
        x = anything
    return x

# error
def bar_stmt_assign(cond: bool) -> Optional[int]:
    if cond:
        x = anything
    else:
        x = None
    return x


##### Conditional statement without assignment

# error
def foo_stmt_return(cond: bool) -> Optional[int]:
    if cond:
        return None
    else:
        return anything  

# error
def bar_stmt_return(cond: bool) -> Optional[int]:
    if cond:
        return anything
    else:
        return None
```

**Expected Behavior**

Each function above should throw the same `no-any-return` error

**Actual Behavior**

Only those marked `# error` throw such an error

**Your Environment**

- Mypy version used: 0.790
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.8.6
- Operating system and version: Windows 10, 64-bit, v.1909

**Comment by hauntsaninja** [comment]
This is the same issue as #9213
